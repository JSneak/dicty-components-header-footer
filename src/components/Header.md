## Default header component

```jsx
const headerTheme = {
  primary: "#004080",
  secondary: "#001b53",
  text: "#004080",
}

const items = [
  {
    url: "/cite",
    icon: "plus",
    text: "Cite Us",
  },
  {
    url: "/downloads",
    icon: "download",
    text: "Downloads",
  },
  {
    url: "/info",
    icon: "info-circle",
    text: "About dictyBase",
  },
  {
    url: "/login",
    icon: "sign-in",
    text: "Login",
    isRouter: true,
  },
]

const style = {
  marginRight: "8px",
  marginTop: "24px",
}

const HeaderLinks = ({ items }) => {
  return (
    <React.Fragment>
      {items.map((link, i) => (
        <a key={i} href={link.url} style={style}>
          {link.text}
        </a>
      ))}
    </React.Fragment>
  )
}

;<Header items={items} render={HeaderLinks} theme={headerTheme} />
```
