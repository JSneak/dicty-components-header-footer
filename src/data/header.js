module.exports = [
  {
    url: "/login",
    icon: "sign-in",
    text: "Login",
  },
  {
    url: "/cite",
    icon: "plus",
    text: "Cite Us",
  },
  {
    url: "/downloads",
    icon: "download",
    text: "Downloads",
    isRouter: true,
  },
  {
    url: "/info",
    icon: "info-circle",
    text: "About dictyBase",
  },
]
