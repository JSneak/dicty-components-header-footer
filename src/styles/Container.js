import React from "react"
import styled from "styled-components"

const ContainerStyle = styled.div`
  width: 100%;
  display: flex;
  flex-direction: row;
`
const Container = ({ children, ...props }) =>
  <ContainerStyle {...props}>
    {children}
  </ContainerStyle>

export default Container
