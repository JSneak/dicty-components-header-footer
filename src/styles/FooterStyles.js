export const FooterStyles = theme => ({
  root: {
    width: "100%",
    fontFamily: "Roboto, sans-serif",
    background: "linear-gradient(#2c3e50, #3498db)",
    paddingLeft: 20,
    paddingRight: 20,
    paddingBottom: 5,
  },
  listItem: {
    margin: 0,
    padding: 1,
    listStyle: "none",
  },
  link: {
    color: "#d8d8d8",
    fontSize: 11,
    textDecoration: "none",
    "&:hover": {
      textDecoration: "underline",
    },
  },
  subsectionItem: {
    paddingLeft: 10,
    paddingRight: 10,
  },
  ulHeader: {
    color: "#19d6a0",
    fontSize: 14,
    textDecoration: "none",
  },
  // center text for all items/links at sm/xs breakpoints
  [theme.breakpoints.down("sm")]: {
    listItem: {
      textAlign: "center",
    },
    headerLink: {
      textAlign: "center",
    },
  },
  // tighten up spacing for xs breakpoints
  [theme.breakpoints.down("xs")]: {
    root: {
      padding: 0,
    },
    gridItem: {
      width: "50%",
    },
    subsectionItem: {
      paddingLeft: 0,
    },
  },
})
