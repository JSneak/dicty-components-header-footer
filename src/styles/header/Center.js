import React from "react"
import styled from "styled-components"

const CenterStyle = styled.div`
  width: 28%;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: flex-end;

  @media only screen and (max-width: 965px) {
    display: flex;
    flex-basis: 100%;
    align-items: flex-end;
    justify-content: center;
    order: initial;
    padding: 0px 5px 5px 0px;
  }
`

const Center = ({ children, ...props }) => (
  <CenterStyle {...props}>{children}</CenterStyle>
)

export default Center
