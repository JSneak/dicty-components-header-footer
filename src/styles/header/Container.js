import React from "react"
import styled from "styled-components"

const ContainerStyle = styled.div`
  display: flex;
  flex-flow: row wrap;
  margin-left: auto;
  margin-right: auto;
  justify-content: space-between;
`
const Container = ({ children, ...props }) => (
  <ContainerStyle {...props}>{children}</ContainerStyle>
)

export default Container
