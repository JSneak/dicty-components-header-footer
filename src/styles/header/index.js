export { Img } from "./Img"
export { Link } from "./Link"
export { LinksContainer } from "./LinksContainer"
export { Logo } from "./Logo"
export { Magnifier } from "./Magnifier"
export { Right } from "./Right"
export { SearchBar } from "./SearchBar"
export { SearchBarContainer } from "./SearchContainer"
export { Container } from "./Container"
