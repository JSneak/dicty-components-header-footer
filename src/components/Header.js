// @flow
import React, { useState } from "react"
import logo from "../images/logo.png"
import Link from "../styles/Link"
import { headerStyles as styles } from "../styles/headerStyles"
import { withStyles } from "@material-ui/core/styles"
import Grid from "@material-ui/core/Grid"
import ExpandedSearch from "./ExpandedSearch"
import NormalSearch from "./NormalSearch"

// set base URL for homepage
const Home = "/"

type ItemType = {
  /** url link */
  url: string,
  /** FontAwesome icon to be displayed */
  icon: string,
  /** description of the link that will be displayed */
  text: string,
}

type Props = {
  /** List of header items, inside an array */
  items: Array<ItemType>,
  /** Link for the logo, that goes to the home page */
  home?: string,
  /** Children react node which renders the links in the header */
  children: any,
  /** Material-UI's classes object for styling */
  classes: Object,
}

/**
 * The `Header` component that will be displayed
 * on top of the navigation bar in every React web application of [dictyBase](http://dictybase.org).
 */

const Header = (props: Props) => {
  const [isExpanded, setIsExpanded] = useState(false)
  const { classes, home = Home, children, items } = props

  const handleClick = () => {
    setIsExpanded(!isExpanded)
  }

  return (
    <Grid container className={classes.root} alignItems="center">
      <Grid item xs={12} sm={5} md={3} lg={3} className={classes.logoContainer}>
        <Link href={home}>
          <img src={logo} alt="dictyBase logo" className={classes.image} />
        </Link>
      </Grid>

      <Grid item className={classes.dcr} xs={12} sm={7} md={5} lg={5}>
        <center>
          <br />
          <em>Dicty Community Resource</em>
        </center>
      </Grid>
      <Grid item xs={12} sm={12} md={4} lg={4}>
        <div className={classes.linkContainer}>{children(items)}</div>
        {isExpanded ? (
          <ExpandedSearch isExpanded={isExpanded} />
        ) : (
          <NormalSearch handleClick={handleClick} />
        )}
      </Grid>
    </Grid>
  )
}

export default withStyles(styles)(Header)
