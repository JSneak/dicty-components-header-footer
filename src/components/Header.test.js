import React from "react"
import Header from "./Header"
import renderer from "react-test-renderer"
import items from "../data/header"
import { Link } from "../styles/header"
import "jest-styled-components"
// $FlowFixMe
import "font-awesome/css/font-awesome.min.css"
import FontAwesome from "react-fontawesome"

const generateLinks = (link, i) => {
  return (
    <Link key={i} href={link.url}>
      <FontAwesome name={link.icon} />
      {link.text}
    </Link>
  )
}

test("matching a snapshot of Header", () => {
  const component = renderer.create(
    <Header items={items}>{items => items.map(generateLinks)}</Header>,
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
