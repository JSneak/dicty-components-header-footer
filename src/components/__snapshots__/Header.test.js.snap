// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of Header 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  padding-right: 15px;
  padding-left: 15px;
  margin-left: auto;
  margin-right: auto;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
}

.c1 {
  width: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  height: 100px;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-flex-basis: calc(33.33333333333333% - 8px);
  -ms-flex-basis: calc(33.33333333333333% - 8px);
  flex-basis: calc(33.33333333333333% - 8px);
}

.c2 {
  color: #15317e;
  margin-right: 8px;
  text-decoration: none;
}

.c3 {
  max-width: 220px;
  height: 78px;
}

.c4 {
  width: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  padding: 20px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin-bottom: 5px;
  -webkit-flex-basis: 100%;
  -ms-flex-basis: 100%;
  flex-basis: 100%;
  font-size: 14px;
}

.c6 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c7 {
  cursor: not-allowed;
  border: 1px solid rgb(204, 204, 204);
  border-top-right-radius: 0px;
  border-bottom-right-radius: 0px;
  border-top-left-radius: 3px;
  border-bottom-left-radius: 3px;
  width: 50%;
  height: 30px;
  color: #555;
  padding: 0px 0px 0px 5px;
}

.c8 {
  height: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0px 7px;
  cursor: not-allowed;
  border: 1px solid rgb(204, 204, 204);
  border-width: 1px 1px 1px 0px;
  border-color: rgb(204, 204, 204);
  border-style: solid;
  border-top-right-radius: 3px;
  border-bottom-right-radius: 3px;
}

@media only screen and (max-width: 800px) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-basis: 100%;
    -ms-flex-basis: 100%;
    flex-basis: 100%;
    -webkit-align-items: flex-end;
    -webkit-box-align: flex-end;
    -ms-flex-align: flex-end;
    align-items: flex-end;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-order: initial;
    -ms-flex-order: initial;
    order: initial;
  }
}

@media only screen and (max-width: 800px) {
  .c4 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-basis: 100%;
    -ms-flex-basis: 100%;
    flex-basis: 100%;
    -webkit-align-items: flex-end;
    -webkit-box-align: flex-end;
    -ms-flex-align: flex-end;
    align-items: flex-end;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-order: initial;
    -ms-flex-order: initial;
    order: initial;
    padding: 0px 5px 5px 0px;
  }
}

@media only screen and (max-width: 800px) {
  .c5 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-basis: 100%;
    -ms-flex-basis: 100%;
    flex-basis: 100%;
    -webkit-align-items: flex-end;
    -webkit-box-align: flex-end;
    -ms-flex-align: flex-end;
    align-items: flex-end;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-order: initial;
    -ms-flex-order: initial;
    order: initial;
  }
}

@media only screen and (max-width: 1024px) {
  .c7 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-basis: 100%;
    -ms-flex-basis: 100%;
    flex-basis: 100%;
    -webkit-align-items: flex-end;
    -webkit-box-align: flex-end;
    -ms-flex-align: flex-end;
    align-items: flex-end;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-order: initial;
    -ms-flex-order: initial;
    order: initial;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <a
      className="c2"
      href="http://dictybase.org"
    >
      <img
        className="c3"
        src="test-file-stub"
      />
    </a>
  </div>
  <div
    className="c4"
  >
    <div
      className="c5"
    >
      <a
        className="c2"
        href="/login"
      >
        <span
          aria-hidden={true}
          className="fa fa-sign-in"
        />
        Login
      </a>
      <a
        className="c2"
        href="/cite"
      >
        <span
          aria-hidden={true}
          className="fa fa-plus"
        />
        Cite Us
      </a>
      <a
        className="c2"
        href="/downloads"
      >
        <span
          aria-hidden={true}
          className="fa fa-download"
        />
        Downloads
      </a>
      <a
        className="c2"
        href="/info"
      >
        <span
          aria-hidden={true}
          className="fa fa-info-circle"
        />
        About dictyBase
      </a>
    </div>
    <div
      className="c6"
    >
      <input
        className="c7"
        disabled={true}
        placeholder="Guided Search (coming soon)"
      />
      <div
        className="c8"
      >
        <span
          aria-hidden={true}
          className="fa fa-search"
        />
      </div>
    </div>
  </div>
</div>
`;
